vmware/build-tools-for-vmware-aria
# [ecmascript] (#361) Add Object Support in the VROES.Shims.arrayFrom()

### Description

Add support of objects in the `VROES.Shims.arrayFrom()` method so its behavior is similar to the standard `Array.from()` method.

### Checklist

- [x] I have updated the PR title with affected component, related issue number and a short summary of the changes introduced
- [x] I have added labels for implementation kind (kind/*) and version type (version/*)

### Testing

1. Create a workflow with one script able task that has the following code:
```js
var __global = System.getContext() || (function () { return this; }).call(null);
var VROES = __global.__VROES || (__global.__VROES = System.getModule("com.vmware.pscoe.library.ecmascript").VROES()), exports = {};

var array1 = array1 = VROES.Shims.arrayFrom({ length: 5 });
System.log("Array 1  data: " + JSON.stringify(array1, null, 4));

var arr = [{ key1: 1 }, { key2: 2 }, 3, "ddd"];
var array2 = VROES.Shims.arrayFrom(arr);
System.log("Array 2  data: " + JSON.stringify(array2, null, 4));
```
2. Run the workflow, the output should be array based on the input object.

#### Test Screenshot

![361_array_test_results](https://github.com/user-attachments/assets/81706f1b-872d-46b5-b6e0-33352d90ef40)

### Release Notes

Add support of objects in the `VROES.Shims.arrayFrom()` method so its behavior is similar to the standard `Array,from()` method.

### Related issues and PRs

Resolves #361
